bootstrap/kubeadm/api/v1alpha3/conversion.go

```diff
@@ -19,19 +19,41 @@ package v1alpha3
 import (
 	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	kubeadmbootstrapv1alpha4 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha4"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 // ConvertTo converts this KubeadmConfig to the Hub version (v1alpha4).
 func (src *KubeadmConfig) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*kubeadmbootstrapv1alpha4.KubeadmConfig)
-	return Convert_v1alpha3_KubeadmConfig_To_v1alpha4_KubeadmConfig(src, dst, nil)
+	if err := Convert_v1alpha3_KubeadmConfig_To_v1alpha4_KubeadmConfig(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &kubeadmbootstrapv1alpha4.KubeadmConfig{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
```

Comment:
If `!ok` this error will return nil?

Reply:
Will mark as resolved as duplicate, so we keep discussion in one place. I'll remember to fix all occurrences if needed.